.changeset/add-no-script-url-rule.md

```diff
@@ -0,0 +1,28 @@
+---
+"@biomejs/biome": patch
+---
+
+Added the rule [`noScriptUrl`](https://biomejs.dev/linter/rules/no-script-url/).
+
+This rule disallows the use of `javascript:` URLs, which are considered a form of `eval` and can pose security risks such as XSS vulnerabilities.
+
+**Invalid code**
+```jsx
+<a href="javascript:void(0)">Click me</a>
+<a href="javascript:alert('XSS')">Click me</a>
+React.createElement('a', { href: 'javascript:void(0)' });
+<Link href="javascript:void(0)">Click me</Link>
+
+**Invalid code**
+```jsx
+<a href="javascript:void(0)">Click me</a>
+<a href="javascript:alert('XSS')">Click me</a>
+React.createElement('a', { href: 'javascript:void(0)' });
+```
+**Valid code**
+```jsx
+<a href="https://example.com">Click me</a>
+<a href="/path/to/page">Click me</a>
+<a href="#section">Click me</a>
+```
```

Comment:
nitpicks:
 
1. duplicated invalid blocks
1. Would prefer to keep it brief. The more vertical space a changeset takes indicates to users that the change is more important and requires more attention. I like the `<a href="javascript:alert('XSS')">Click me</a>` example as the invalid case and the `<a href="https://example.com">Click me</a>` as the valid case. Personally, I would go as far as to omit the valid cases altogether, but no strong feelings, up to you.